src/Shared/UnitTests/TypeLoader_Dependencies_Tests.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+using System;
+using System.IO;
+using Microsoft.Build.Shared;
+using Microsoft.Build.SharedUtilities;
+using Xunit;
+
+namespace Microsoft.Build.UnitTests
+{
+    public class TypeLoader_Dependencies_Tests
+    {
+        private static readonly string ProjectFilePath = Path.Combine(BuildEnvironmentHelper.Instance.CurrentMSBuildToolsDirectory, "TaskWithDependencyTest.proj");
+        private static readonly string TaskDllFileName = "TaskWithDependency.dll";
+        private static readonly string OriginalTaskDllPath = Path.Combine(BuildEnvironmentHelper.Instance.CurrentMSBuildToolsDirectory, TaskDllFileName);
+        private static readonly string DependencyDllFileName = "Dependency.dll";
+        private static readonly string OriginalDependencyDllPath = Path.Combine(BuildEnvironmentHelper.Instance.CurrentMSBuildToolsDirectory, DependencyDllFileName);
+
+        [Fact]
+        public void LoadAssemblyAndDependency_Inside()
```

Comment:
Can you be a bit more explicit than "inside"/"outside"? I'm not sure a failure with this test name would be clear to me.